token/client/src/token.rs

```diff
@@ -388,13 +388,12 @@ where
     fn get_multisig_signers<'a, 'b>(
         &self,
         authority: &'b Pubkey,
-        signing_pubkeys: &'a Vec<Pubkey>,
+        signing_pubkeys: &'a [Pubkey],
     ) -> Vec<&'a Pubkey> {
-        if signing_pubkeys.as_ref() == [*authority] {
-            vec![]
-        } else {
-            signing_pubkeys.iter().collect::<Vec<_>>()
-        }
```

Comment:
I can't remember why we decided to special-case this, but it seems like we should always filter the authority from the multisig pubkeys, right? This breaks a self-owned multisig, ie a multisig where the address is also a participant. I'm not even sure if that's supported in the program to be honest.

Reply:
Never mind, I figured it out and reverted it! If it's self-owned, we don't want to remove `authority` from this list. But also, we need to work out nonces. If there's a nonce authority in the bulk signers, we need to make sure it's properly omitted here. Gonna think about this one

Reply:
i dont remember exactly how this interacted with nonce mode, but (iirc, that is, i know this is true but i dont know if there are other special cases im forgetting), the idea of this was...

originally, basically every signer got pushed into `bulk_signers`, which meant every token instruction received every transaction signer as a possible multisig signer. to deal with this, first we stopped pushing duplicates into the array. then we changed `payer` on `Token` to be a signer rather than a pubkey, so that we could exclude it from `bulk_signers` too. i _think_ at this point that meant the vec could only contain:
* one authority for a single-signer instruction
* multiple authorities for a multi-signer instruction
* one authority for the instruction plus the nonce authority in nonce mode

and distinguishing the second and third cases was not possible inside the token client, so.... ok yes i think this is all correct and covers the last case that i didnt

the detail of how all this came about is in #3509 and #3649 fwiw